### ISSUE TITLE
Upgrade from PS 1.7.6.8 to PS 1.7.6.9 downgrades blockreassurance module version

### ISSUE THREAD

**Issue by photinia73** [PrestaShop/PrestaShop#21982]
**Describe your problem**
Ufter updating PS to 1.7.6.9 version the blockreassurance module was repkaced by its previos version (without URL).
Then it ask the update in the BO
After updating I saw that in the DB tables ps_psreassurance and ps_psreassurance_lang was replaced with default (( 
I restored them from backup, but this 

My site https://old.te.com.ua this is the clone of https://te.com.ua/uk/ and I update it at first
The same thing was aftre updating PS to 1.7.6.8....  I can`t understand why this happend

**Steps to reproduce**

1. Install PS 1.7.6.8
2. Upgrade blockreassurance module to latest version v5.0.0
3. Upgrade PrestaShop to PS 1.7.6.9
4. Module blockreassurance has been downgraded to v4

**Screenshots**

https://drive.google.com/file/d/1M3Kkr6kXzdsehRWLmBTTn2ad_EzaCI08/view

After the upgrade:

https://drive.google.com/file/d/1MI1AwMBotvAGCJQO60lRCLhd7seAFL_-/view

#### Additional information
 PS 1.7.6.9
PHP 7.1.33-16+0~20200514.38+debian8~1.gbp1e5820
MySQL: 10.0.38-MariaDB-1~jessie
Theme HiddenTechies PixtronPro (here was old .tpl of blockreassurance)

**Comment by matks** [comment]
This might indicate that, upon upgrade, native modules are restored to the version of the ZIP.
If prestashop1.7.6.9.zip contains v4.1.1 but your shop uses v5.0.0, the module would replace it?

To be verified

**Comment by photinia73** [comment]
No. In BO it look like 
![image](https://user-images.githubusercontent.com/50954519/99548484-5ebfff00-29c1-11eb-8f2d-8805b164bf64.png)

and inside module
![image](https://user-images.githubusercontent.com/50954519/99548641-89aa5300-29c1-11eb-9ae7-4478415fd2ff.png)

**Comment by photinia73** [comment]
But after updating this module rollback to the version without URL , unfortunately I didn`t pay attantion 4.1.0 or 4.0.1 or 4.1.1 
And it erase the tables in the DB ((

**Comment by photinia73** [comment]
This is not a critical problem if you have backup )), but take a time and some nerves ))

**Comment by hibatallahAouadni** [comment]
Hello @photinia73

Thanks for your issue. **You must follow the template to submit a complete bug report**.
Read more about how we expect the issues to be handled [here](https://build.prestashop.com/news/how-bug-reports-are-handled/). Please edit your issue, that we help us to investigate your problem.

Thanks!

**Comment by photinia73** [comment]
Corrected. I find in the my Theme in Modules blockreassurance/view/hook/....tpl may be the problem from this case....

**Comment by hibatallahAouadni** [comment]
@photinia73 so the problem is not a PrestaShop Core bug but it's from your side, is that so?

**Comment by photinia73** [comment]
I am not shure... I made the changes in the Theme and will update PS on the main site (not clone for test). After this I will know exactly

**Comment by hibatallahAouadni** [comment]
ok! So, please check and feedback.
Thanks!

**Comment by photinia73** [comment]
Of cource

**Comment by lacell75** [comment]
After updating PS to 1.7.6.9 version, block reassurance is broken in my side also. I have seen that the tables have been renamed (ps_reassurance to reassurance), please see screenshots attached
![reassurance_BDD_rename](https://user-images.githubusercontent.com/33258660/99903755-bd89bf00-2cc6-11eb-961b-b83c3238b603.png)
![reassurance_broken](https://user-images.githubusercontent.com/33258660/99903758-c1b5dc80-2cc6-11eb-9e4e-fa43795a09d1.png)

**Comment by matks** [comment]
@lacell75 Would this be an accurate description of your issue ?

Steps to reproduce the behavior:

1. Install PS 1.7.6.8
2. Upgrade blockreassurance module to latest version v5.0.0
3. Upgrade PrestaShop to PS 1.7.6.9
4. Module blockreassurance has been downgraded to v4

Can you confirm?

**Comment by lacell75** [comment]
Yes I confirm. I had the latest version V5.0.0 before the update then 4.1.1.

**Comment by photinia73** [comment]
You are right. 
After upgrade to 1.7.6.9 module blockreassurance have new tables (( And these tables have no fields Link, Description etc…)  
I think this is not only my problem
My version of blockreassurance is 4.1.1 Do  blockreassurance 5.0.0 have thr same DB fields as v 4.1.1? link, description possibilities?

**Comment by matks** [comment]
> You are right. After upgrade to 1.7.6.9 module blockreassurance have new tables (( And these tables have no fields Link, Description etc…)  I think this is not only my problem My version of blockreassurance is 4.1.1 Do  blockreassurance 5.0.0 have thr same DB fields as v 4.1.1? link, description possibilities?

v5.0.0 has the same global behavior than v4.1.1 (same database structure) plus some bug fixes and new features, you can see changelog here https://github.com/PrestaShop/blockreassurance/releases

**Comment by hibatallahAouadni** [comment]
Hello @photinia73 

Following your steps with PS1.7.6.8, here's what I got (see the attached screen records below):

https://drive.google.com/file/d/1M3Kkr6kXzdsehRWLmBTTn2ad_EzaCI08/view

After the upgrade:

https://drive.google.com/file/d/1MI1AwMBotvAGCJQO60lRCLhd7seAFL_-/view

It's weird! in Modules Manager it's mentionned that the Customer reassurance module is v5.0.0, then when we click on Configure button, the old template of the module is displayed!!

@matks please check and feedback.

Thanks!

**Comment by photinia73** [comment]
Exactly!

**Comment by matks** [comment]
> It's weird! in Modules Manager it's mentionned that the Customer reassurance module is v5.0.0, then when we click on Configure button, the old template of the module is displayed!!

This might be because PrestaShop "does not know" that module version was downgraded. I think actually what happens is that the upgrade process has replaced the source code files, but has not applied a "proper" downgrade process.

**Comment by hibatallahAouadni** [comment]
Hello @photinia73 ,

Well then, it's confirmed! Thanks @matks 
I’ll add this to the debug backlog so that it’s fixed.
If you have already fixed it on your end or if you think you can do it, please do send us a pull request!

Thank you

**Comment by photinia73** [comment]
I would be glad to do something useful, but I`m not developer (( Thank you!

**Comment by damianobertuna** [comment]
Hi everybody, watch this video where I explain how to have again the module working:
[video solution](https://youtu.be/o5zNZTV4D1A)

As I said it's not a solution for the bug, but a tip to have again the module working on the website.

**Comment by PierreRambaud** [comment]
FYI, it's in the TODO list of the autoupgrade: https://github.com/PrestaShop/autoupgrade/blob/master/classes/TaskRunner/Upgrade/UpgradeFiles.php#L58
We are open for pull request :)
